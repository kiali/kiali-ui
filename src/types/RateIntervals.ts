import { config } from '../config';

const mapIntervals: { [key: number]: string } = config().toolbar.intervalDuration;
export const tuples: [number, string][] = Object.keys(mapIntervals).map(key => {
  const tuple: [number, string] = [+key, mapIntervals[key]];
  return tuple;
});

export const getName = (durationSeconds: number): string => {
  const name = mapIntervals[durationSeconds];
  if (name) {
    return name;
  }
  return 'Last ' + durationSeconds + ' seconds';
};
