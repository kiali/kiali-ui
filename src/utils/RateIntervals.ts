import { serverConfig } from '../config/ServerConfig';

export const getName = (durationSeconds: number): string => {
  const name = serverConfig.durations[durationSeconds];
  if (name) {
    return name;
  }
  return durationSeconds + ' seconds';
};
