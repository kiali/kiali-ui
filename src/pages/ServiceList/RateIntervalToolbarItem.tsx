import * as React from 'react';
import { DropdownButton, MenuItem } from 'patternfly-react';
import RateIntervals from '../../types/RateIntervals';

type RateIntervalToolbarItemProps = {
  rateIntervalSelected: string;
  onRateIntervalChanged?: (key: string) => void;
};

export default class RateIntervalToolbarItem extends React.Component<RateIntervalToolbarItemProps> {
  render() {
    const rateIntervalSelected = RateIntervals.find(el => {
      return el[0] === this.props.rateIntervalSelected;
    });

    return (
      <div className="form-group">
        <label style={{ paddingRight: '0.5em' }}>Rate Interval:</label>
        <DropdownButton
          title={'Last ' + rateIntervalSelected![1]}
          onSelect={this.props.onRateIntervalChanged}
          id="rateIntervalDropDown"
        >
          {RateIntervals.map(r => (
            <MenuItem key={r[0]} active={r[0] === this.props.rateIntervalSelected} eventKey={r[0]}>
              Last {r[1]}
            </MenuItem>
          ))}
        </DropdownButton>
      </div>
    );
  }
}
