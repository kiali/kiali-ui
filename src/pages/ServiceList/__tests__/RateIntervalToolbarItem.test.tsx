import * as React from 'react';
import { shallow } from 'enzyme';
import RateIntervalToolbarItem from '../RateIntervalToolbarItem';
import RateIntervals from '../../../types/RateIntervals';

describe('RateIntervalToolbarItem', () => {
  it('renders correctly', () => {
    const wrapper = shallow(<RateIntervalToolbarItem rateIntervalSelected="10m" />);
    expect(wrapper).toBeDefined();
    expect(wrapper).toMatchSnapshot();
    expect(wrapper.find('MenuItem').length).toBe(RateIntervals.length);
  });
});
