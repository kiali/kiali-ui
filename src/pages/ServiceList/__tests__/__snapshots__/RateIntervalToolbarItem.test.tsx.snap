// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RateIntervalToolbarItem renders correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <RateIntervalToolbarItem
    rateIntervalSelected={600}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <label
          style={
            Object {
              "paddingRight": "0.5em",
            }
          }
        >
          Rate Interval:
        </label>,
        <DropdownButton
          id="rateIntervalDropDown"
          onSelect={undefined}
          title="Last 10 min"
        >
          <MenuItem
            active={false}
            bsClass="dropdown"
            disabled={false}
            divider={false}
            eventKey={60}
            header={false}
          >
            Last min
          </MenuItem>
          <MenuItem
            active={false}
            bsClass="dropdown"
            disabled={false}
            divider={false}
            eventKey={300}
            header={false}
          >
            Last 5 min
          </MenuItem>
          <MenuItem
            active={true}
            bsClass="dropdown"
            disabled={false}
            divider={false}
            eventKey={600}
            header={false}
          >
            Last 10 min
          </MenuItem>
          <MenuItem
            active={false}
            bsClass="dropdown"
            disabled={false}
            divider={false}
            eventKey={1800}
            header={false}
          >
            Last 30 min
          </MenuItem>
          <MenuItem
            active={false}
            bsClass="dropdown"
            disabled={false}
            divider={false}
            eventKey={3600}
            header={false}
          >
            Last hour
          </MenuItem>
          <MenuItem
            active={false}
            bsClass="dropdown"
            disabled={false}
            divider={false}
            eventKey={10800}
            header={false}
          >
            Last 3 hours
          </MenuItem>
          <MenuItem
            active={false}
            bsClass="dropdown"
            disabled={false}
            divider={false}
            eventKey={21600}
            header={false}
          >
            Last 6 hours
          </MenuItem>
          <MenuItem
            active={false}
            bsClass="dropdown"
            disabled={false}
            divider={false}
            eventKey={43200}
            header={false}
          >
            Last 12 hours
          </MenuItem>
          <MenuItem
            active={false}
            bsClass="dropdown"
            disabled={false}
            divider={false}
            eventKey={86400}
            header={false}
          >
            Last day
          </MenuItem>
          <MenuItem
            active={false}
            bsClass="dropdown"
            disabled={false}
            divider={false}
            eventKey={604800}
            header={false}
          >
            Last 7 days
          </MenuItem>
          <MenuItem
            active={false}
            bsClass="dropdown"
            disabled={false}
            divider={false}
            eventKey={2592000}
            header={false}
          >
            Last 30 days
          </MenuItem>
        </DropdownButton>,
      ],
      "className": "form-group",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Rate Interval:",
          "style": Object {
            "paddingRight": "0.5em",
          },
        },
        "ref": null,
        "rendered": "Rate Interval:",
        "type": "label",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "children": Array [
            <MenuItem
              active={false}
              bsClass="dropdown"
              disabled={false}
              divider={false}
              eventKey={60}
              header={false}
            >
              Last min
            </MenuItem>,
            <MenuItem
              active={false}
              bsClass="dropdown"
              disabled={false}
              divider={false}
              eventKey={300}
              header={false}
            >
              Last 5 min
            </MenuItem>,
            <MenuItem
              active={true}
              bsClass="dropdown"
              disabled={false}
              divider={false}
              eventKey={600}
              header={false}
            >
              Last 10 min
            </MenuItem>,
            <MenuItem
              active={false}
              bsClass="dropdown"
              disabled={false}
              divider={false}
              eventKey={1800}
              header={false}
            >
              Last 30 min
            </MenuItem>,
            <MenuItem
              active={false}
              bsClass="dropdown"
              disabled={false}
              divider={false}
              eventKey={3600}
              header={false}
            >
              Last hour
            </MenuItem>,
            <MenuItem
              active={false}
              bsClass="dropdown"
              disabled={false}
              divider={false}
              eventKey={10800}
              header={false}
            >
              Last 3 hours
            </MenuItem>,
            <MenuItem
              active={false}
              bsClass="dropdown"
              disabled={false}
              divider={false}
              eventKey={21600}
              header={false}
            >
              Last 6 hours
            </MenuItem>,
            <MenuItem
              active={false}
              bsClass="dropdown"
              disabled={false}
              divider={false}
              eventKey={43200}
              header={false}
            >
              Last 12 hours
            </MenuItem>,
            <MenuItem
              active={false}
              bsClass="dropdown"
              disabled={false}
              divider={false}
              eventKey={86400}
              header={false}
            >
              Last day
            </MenuItem>,
            <MenuItem
              active={false}
              bsClass="dropdown"
              disabled={false}
              divider={false}
              eventKey={604800}
              header={false}
            >
              Last 7 days
            </MenuItem>,
            <MenuItem
              active={false}
              bsClass="dropdown"
              disabled={false}
              divider={false}
              eventKey={2592000}
              header={false}
            >
              Last 30 days
            </MenuItem>,
          ],
          "id": "rateIntervalDropDown",
          "onSelect": undefined,
          "title": "Last 10 min",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": "60",
            "nodeType": "class",
            "props": Object {
              "active": false,
              "bsClass": "dropdown",
              "children": "Last min",
              "disabled": false,
              "divider": false,
              "eventKey": 60,
              "header": false,
            },
            "ref": null,
            "rendered": "Last min",
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": "300",
            "nodeType": "class",
            "props": Object {
              "active": false,
              "bsClass": "dropdown",
              "children": "Last 5 min",
              "disabled": false,
              "divider": false,
              "eventKey": 300,
              "header": false,
            },
            "ref": null,
            "rendered": "Last 5 min",
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": "600",
            "nodeType": "class",
            "props": Object {
              "active": true,
              "bsClass": "dropdown",
              "children": "Last 10 min",
              "disabled": false,
              "divider": false,
              "eventKey": 600,
              "header": false,
            },
            "ref": null,
            "rendered": "Last 10 min",
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": "1800",
            "nodeType": "class",
            "props": Object {
              "active": false,
              "bsClass": "dropdown",
              "children": "Last 30 min",
              "disabled": false,
              "divider": false,
              "eventKey": 1800,
              "header": false,
            },
            "ref": null,
            "rendered": "Last 30 min",
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": "3600",
            "nodeType": "class",
            "props": Object {
              "active": false,
              "bsClass": "dropdown",
              "children": "Last hour",
              "disabled": false,
              "divider": false,
              "eventKey": 3600,
              "header": false,
            },
            "ref": null,
            "rendered": "Last hour",
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": "10800",
            "nodeType": "class",
            "props": Object {
              "active": false,
              "bsClass": "dropdown",
              "children": "Last 3 hours",
              "disabled": false,
              "divider": false,
              "eventKey": 10800,
              "header": false,
            },
            "ref": null,
            "rendered": "Last 3 hours",
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": "21600",
            "nodeType": "class",
            "props": Object {
              "active": false,
              "bsClass": "dropdown",
              "children": "Last 6 hours",
              "disabled": false,
              "divider": false,
              "eventKey": 21600,
              "header": false,
            },
            "ref": null,
            "rendered": "Last 6 hours",
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": "43200",
            "nodeType": "class",
            "props": Object {
              "active": false,
              "bsClass": "dropdown",
              "children": "Last 12 hours",
              "disabled": false,
              "divider": false,
              "eventKey": 43200,
              "header": false,
            },
            "ref": null,
            "rendered": "Last 12 hours",
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": "86400",
            "nodeType": "class",
            "props": Object {
              "active": false,
              "bsClass": "dropdown",
              "children": "Last day",
              "disabled": false,
              "divider": false,
              "eventKey": 86400,
              "header": false,
            },
            "ref": null,
            "rendered": "Last day",
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": "604800",
            "nodeType": "class",
            "props": Object {
              "active": false,
              "bsClass": "dropdown",
              "children": "Last 7 days",
              "disabled": false,
              "divider": false,
              "eventKey": 604800,
              "header": false,
            },
            "ref": null,
            "rendered": "Last 7 days",
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": "2592000",
            "nodeType": "class",
            "props": Object {
              "active": false,
              "bsClass": "dropdown",
              "children": "Last 30 days",
              "disabled": false,
              "divider": false,
              "eventKey": 2592000,
              "header": false,
            },
            "ref": null,
            "rendered": "Last 30 days",
            "type": [Function],
          },
        ],
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <label
            style={
              Object {
                "paddingRight": "0.5em",
              }
            }
          >
            Rate Interval:
          </label>,
          <DropdownButton
            id="rateIntervalDropDown"
            onSelect={undefined}
            title="Last 10 min"
          >
            <MenuItem
              active={false}
              bsClass="dropdown"
              disabled={false}
              divider={false}
              eventKey={60}
              header={false}
            >
              Last min
            </MenuItem>
            <MenuItem
              active={false}
              bsClass="dropdown"
              disabled={false}
              divider={false}
              eventKey={300}
              header={false}
            >
              Last 5 min
            </MenuItem>
            <MenuItem
              active={true}
              bsClass="dropdown"
              disabled={false}
              divider={false}
              eventKey={600}
              header={false}
            >
              Last 10 min
            </MenuItem>
            <MenuItem
              active={false}
              bsClass="dropdown"
              disabled={false}
              divider={false}
              eventKey={1800}
              header={false}
            >
              Last 30 min
            </MenuItem>
            <MenuItem
              active={false}
              bsClass="dropdown"
              disabled={false}
              divider={false}
              eventKey={3600}
              header={false}
            >
              Last hour
            </MenuItem>
            <MenuItem
              active={false}
              bsClass="dropdown"
              disabled={false}
              divider={false}
              eventKey={10800}
              header={false}
            >
              Last 3 hours
            </MenuItem>
            <MenuItem
              active={false}
              bsClass="dropdown"
              disabled={false}
              divider={false}
              eventKey={21600}
              header={false}
            >
              Last 6 hours
            </MenuItem>
            <MenuItem
              active={false}
              bsClass="dropdown"
              disabled={false}
              divider={false}
              eventKey={43200}
              header={false}
            >
              Last 12 hours
            </MenuItem>
            <MenuItem
              active={false}
              bsClass="dropdown"
              disabled={false}
              divider={false}
              eventKey={86400}
              header={false}
            >
              Last day
            </MenuItem>
            <MenuItem
              active={false}
              bsClass="dropdown"
              disabled={false}
              divider={false}
              eventKey={604800}
              header={false}
            >
              Last 7 days
            </MenuItem>
            <MenuItem
              active={false}
              bsClass="dropdown"
              disabled={false}
              divider={false}
              eventKey={2592000}
              header={false}
            >
              Last 30 days
            </MenuItem>
          </DropdownButton>,
        ],
        "className": "form-group",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Rate Interval:",
            "style": Object {
              "paddingRight": "0.5em",
            },
          },
          "ref": null,
          "rendered": "Rate Interval:",
          "type": "label",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": Array [
              <MenuItem
                active={false}
                bsClass="dropdown"
                disabled={false}
                divider={false}
                eventKey={60}
                header={false}
              >
                Last min
              </MenuItem>,
              <MenuItem
                active={false}
                bsClass="dropdown"
                disabled={false}
                divider={false}
                eventKey={300}
                header={false}
              >
                Last 5 min
              </MenuItem>,
              <MenuItem
                active={true}
                bsClass="dropdown"
                disabled={false}
                divider={false}
                eventKey={600}
                header={false}
              >
                Last 10 min
              </MenuItem>,
              <MenuItem
                active={false}
                bsClass="dropdown"
                disabled={false}
                divider={false}
                eventKey={1800}
                header={false}
              >
                Last 30 min
              </MenuItem>,
              <MenuItem
                active={false}
                bsClass="dropdown"
                disabled={false}
                divider={false}
                eventKey={3600}
                header={false}
              >
                Last hour
              </MenuItem>,
              <MenuItem
                active={false}
                bsClass="dropdown"
                disabled={false}
                divider={false}
                eventKey={10800}
                header={false}
              >
                Last 3 hours
              </MenuItem>,
              <MenuItem
                active={false}
                bsClass="dropdown"
                disabled={false}
                divider={false}
                eventKey={21600}
                header={false}
              >
                Last 6 hours
              </MenuItem>,
              <MenuItem
                active={false}
                bsClass="dropdown"
                disabled={false}
                divider={false}
                eventKey={43200}
                header={false}
              >
                Last 12 hours
              </MenuItem>,
              <MenuItem
                active={false}
                bsClass="dropdown"
                disabled={false}
                divider={false}
                eventKey={86400}
                header={false}
              >
                Last day
              </MenuItem>,
              <MenuItem
                active={false}
                bsClass="dropdown"
                disabled={false}
                divider={false}
                eventKey={604800}
                header={false}
              >
                Last 7 days
              </MenuItem>,
              <MenuItem
                active={false}
                bsClass="dropdown"
                disabled={false}
                divider={false}
                eventKey={2592000}
                header={false}
              >
                Last 30 days
              </MenuItem>,
            ],
            "id": "rateIntervalDropDown",
            "onSelect": undefined,
            "title": "Last 10 min",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": "60",
              "nodeType": "class",
              "props": Object {
                "active": false,
                "bsClass": "dropdown",
                "children": "Last min",
                "disabled": false,
                "divider": false,
                "eventKey": 60,
                "header": false,
              },
              "ref": null,
              "rendered": "Last min",
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": "300",
              "nodeType": "class",
              "props": Object {
                "active": false,
                "bsClass": "dropdown",
                "children": "Last 5 min",
                "disabled": false,
                "divider": false,
                "eventKey": 300,
                "header": false,
              },
              "ref": null,
              "rendered": "Last 5 min",
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": "600",
              "nodeType": "class",
              "props": Object {
                "active": true,
                "bsClass": "dropdown",
                "children": "Last 10 min",
                "disabled": false,
                "divider": false,
                "eventKey": 600,
                "header": false,
              },
              "ref": null,
              "rendered": "Last 10 min",
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": "1800",
              "nodeType": "class",
              "props": Object {
                "active": false,
                "bsClass": "dropdown",
                "children": "Last 30 min",
                "disabled": false,
                "divider": false,
                "eventKey": 1800,
                "header": false,
              },
              "ref": null,
              "rendered": "Last 30 min",
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": "3600",
              "nodeType": "class",
              "props": Object {
                "active": false,
                "bsClass": "dropdown",
                "children": "Last hour",
                "disabled": false,
                "divider": false,
                "eventKey": 3600,
                "header": false,
              },
              "ref": null,
              "rendered": "Last hour",
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": "10800",
              "nodeType": "class",
              "props": Object {
                "active": false,
                "bsClass": "dropdown",
                "children": "Last 3 hours",
                "disabled": false,
                "divider": false,
                "eventKey": 10800,
                "header": false,
              },
              "ref": null,
              "rendered": "Last 3 hours",
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": "21600",
              "nodeType": "class",
              "props": Object {
                "active": false,
                "bsClass": "dropdown",
                "children": "Last 6 hours",
                "disabled": false,
                "divider": false,
                "eventKey": 21600,
                "header": false,
              },
              "ref": null,
              "rendered": "Last 6 hours",
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": "43200",
              "nodeType": "class",
              "props": Object {
                "active": false,
                "bsClass": "dropdown",
                "children": "Last 12 hours",
                "disabled": false,
                "divider": false,
                "eventKey": 43200,
                "header": false,
              },
              "ref": null,
              "rendered": "Last 12 hours",
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": "86400",
              "nodeType": "class",
              "props": Object {
                "active": false,
                "bsClass": "dropdown",
                "children": "Last day",
                "disabled": false,
                "divider": false,
                "eventKey": 86400,
                "header": false,
              },
              "ref": null,
              "rendered": "Last day",
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": "604800",
              "nodeType": "class",
              "props": Object {
                "active": false,
                "bsClass": "dropdown",
                "children": "Last 7 days",
                "disabled": false,
                "divider": false,
                "eventKey": 604800,
                "header": false,
              },
              "ref": null,
              "rendered": "Last 7 days",
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": "2592000",
              "nodeType": "class",
              "props": Object {
                "active": false,
                "bsClass": "dropdown",
                "children": "Last 30 days",
                "disabled": false,
                "divider": false,
                "eventKey": 2592000,
                "header": false,
              },
              "ref": null,
              "rendered": "Last 30 days",
              "type": [Function],
            },
          ],
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
